### DISCUSSION TITLE
Automatically generate Kotlin data classes from Schema

### DISCUSSION THREAD

**Discussion by siddheshlatkar** [ExpediaGroup/graphql-kotlin#1887] [General]
Is there a utility available to generate Kotlin data classes for ExpediaGroup servers from a GraphQL schema?

Why would we require this utility?
Our repository contains common types in the format of a GraphQL schema. In order to support consumers who utilize the code-first approach through the ExpediaGroup Kotlin library, we need to convert all the pre-existing common types into data classes that are compliant with ExpediaGroup's Kotlin server standards.

**Comment by dariuszkuc** [comment]
Hello 👋 
`graphql-kotlin` is a code-first solution that provide Gradle+Maven build plugins to generate schema artifacts. We don't provide any mechanism to build a GraphQL schema from SDL file.

You definitely could write your own plugin to do so (e.g. you could use something like KotlinPoet which we use for our client generator)... but it might be simpler to use Kotlin classes as source of truth and generate schema out of it instead.
